src/app.types.tsx

```diff
@@ -53,31 +53,46 @@ export type DataType = 'scalar' | 'image' | 'waveform';
 
 export interface FullCommonChannelMetadata {
   systemName: string;
-  channel_dtype: DataType;
-  userFriendlyName?: string;
+  type: DataType;
+  path: string;
+  name?: string;
   description?: string;
-  units?: string;
+  historical?: boolean;
 }
 
 export interface FullScalarChannelMetadata extends FullCommonChannelMetadata {
-  channel_dtype: 'scalar';
-  significantFigures?: number;
-  scientificNotation?: boolean;
+  type: 'scalar';
+  precision?: number;
+  notation?: 'scientific' | 'normal';
+  units?: string;
 }
 
 export interface FullImageChannelMetadata extends FullCommonChannelMetadata {
-  channel_dtype: 'image';
+  type: 'image';
 }
 
 export interface FullWaveformChannelMetadata extends FullCommonChannelMetadata {
-  channel_dtype: 'waveform';
+  type: 'waveform';
+  x_units?: string;
+  y_units?: string;
 }
 
 export type FullChannelMetadata =
   | FullScalarChannelMetadata
   | FullImageChannelMetadata
   | FullWaveformChannelMetadata;
 
+// Type guards because TS can't deal with nested discriminated unions
+export const isChannelMetadataScalar = (
+  c: FullChannelMetadata
+): c is FullScalarChannelMetadata => c.type === 'scalar';
+export const isChannelMetadataImage = (
```

Comment:
This seems to be unused. Will this be utilized in the future?

Reply:
It likely will be - I just created TS type assertion functions for all three since they're all related. We haven't had to assert specifically that a channel/it's metadata is an image yet, but likely will when we start doing image manipulation in a later milestone